web-prover-core/src/manifest.rs

```diff
@@ -266,12 +336,182 @@ mod tests {
+      request!(
+          method: "INVALID_METHOD".to_string(),
+          url: "ftp://invalid-scheme.com".to_string(),
+          headers: HashMap::from([
+              ("Authorization".to_string(), "Bearer <% MISSING_TOKEN %>".to_string()),
+              ("Content-Type".to_string(), "invalid/type".to_string())
+          ]),
+          vars: vars
+      ),
+      response!(
+          status: "599".to_string(), // Invalid status
+          headers: HashMap::from([
+              ("Content-Type".to_string(), "invalid/content-type".to_string())
+          ]),
+          body: ManifestResponseBody(ExtractorConfig {
+              format: DataFormat::Json,
+              extractors: vec![
+                  extractor! {
+                      id: "invalid_extractor".to_string(),
+                      description: "This will fail".to_string(),
+                      selector: vec!["invalid".to_string(), "$.".to_string(), "path".to_string()],
+                      extractor_type: ExtractorType::String,
+                  },
+                  extractor! {
+                      id: "another_invalid".to_string(),
+                      description: "This will also fail".to_string(),
+                      selector: vec!["[5]".to_string(), "outofbounds".to_string()],
+                      extractor_type: ExtractorType::Number,
+                  }
+              ],
+          })
+      ),
+    );
+
+    // Validate the manifest
+    let request = request!(
+        method: "POST".to_string(),
+        url: "https://api.example.com".to_string(),
+        version: "HTTP/1.1".to_string(),
+        headers: HashMap::from([
+            ("Authorization".to_string(), "Bearer invalid-token".to_string()),
+            ("Content-Type".to_string(), "application/json".to_string())
+        ]),
+    );
+    let response = NotaryResponse {
+      response:             manifest.response.clone(),
+      notary_response_body: NotaryResponseBody { body: None },
+    };
+    let result = manifest.validate_with(&request, &response).expect("validation should not fail");
+
+    // Verify we get multiple validation errors
+    let mut all_errors = result.errors();
+    all_errors.sort();
+    let mut expected = vec![
+      "Invalid manifest: Invalid HTTP method",
+      "Invalid manifest: Unsupported HTTP status",
+      "Invalid manifest: Token `<% MISSING_TOKEN %>` not declared in `vars`",
+      "Manifest HTTP error: HTTP header mismatch: expected Bearer <% MISSING_TOKEN %>, actual Bearer invalid-token",
+      "Manifest HTTP error: HTTP header mismatch: expected invalid/type, actual application/json",
+      "Template error: Variable missing for key: UNUSED_TOKEN",
+      "Manifest HTTP error: HTTP method mismatch: expected POST, actual INVALID_METHOD",
+      "Manifest HTTP error: HTTP URL mismatch: expected https://api.example.com, actual ftp://invalid-scheme.com",
+      "Extraction failed: No data to match against"
+    ];
```

Comment:
This is the kind of feedback the end-user can expect to receive from the notary. 

cc: @drewjenkins